pkg/provider/pulumi/azure/azure.go

```diff
@@ -0,0 +1,179 @@
+// Copyright Nitric Pty Ltd.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azure
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/pulumi/pulumi-azure/sdk/v4/go/azure/core"
+	"github.com/pulumi/pulumi-azure/sdk/v4/go/azure/eventgrid"
+	"github.com/pulumi/pulumi/sdk/v3/go/auto"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+
+	"github.com/nitrictech/newcli/pkg/provider/pulumi/common"
+	"github.com/nitrictech/newcli/pkg/stack"
+	"github.com/nitrictech/newcli/pkg/target"
+)
+
+type azureProvider struct {
+	s      *stack.Stack
+	t      *target.Target
+	tmpDir string
+}
+
+func New(s *stack.Stack, t *target.Target) common.PulumiProvider {
+	return &azureProvider{s: s, t: t}
+}
+
+func (a *azureProvider) Plugins() []common.Plugin {
+	return []common.Plugin{
+		{
+			Name:    "azure-native",
+			Version: "v1.56.0",
+		},
+		{
+			Name:    "azure",
+			Version: "v4.37.0",
+		},
+		{
+			Name:    "azuread",
+			Version: "v5.16.0",
+		},
+	}
+}
+
+func (a *azureProvider) SupportedRegions() []string {
+	return []string{
+		"westcentralus",
+		"eastus",
+		"westeurope",
+		"jioindiawest",
+		"northeurope",
+		"canadacentral",
+		"eastus2",
+	}
```

Comment:
It looks like with container apps, we may be limited to eastus2 for now: https://azure.microsoft.com/en-au/global-infrastructure/services/?products=container-apps